Is TestForwardPorts test flaky?

I0407 16:48:52.984645   18026 portforward.go:228] Forwarding from 6000 -> 6000
--- FAIL: TestForwardPorts (0.02s)
	portforward_test.go:289: 2: Error reading data: read tcp 127.0.0.1:5000: connection reset by peer
FAIL
coverage: 57.4% of statements
FAIL	 github.com/GoogleCloudPlatform/kubernetes/pkg/client/portforward	0.168s

https://travis-ci.org/GoogleCloudPlatform/kubernetes/jobs/57509181

--- a/pkg/client/portforward/portforward_test.go
+++ b/pkg/client/portforward/portforward_test.go
@@ -138,7 +138,10 @@ func (c *fakeUpgradeConnection) CreateStream(headers http.Header) (httpstream.St
 
 	stream := &fakeUpgradeStream{}
 	c.streams[headers.Get(api.PortHeader)] = stream
-	stream.data = c.portData[headers.Get(api.PortHeader)]
+	// only simulate data on the data stream for now, not the error stream
+	if headers.Get(api.StreamType) == api.StreamTypeData {
+		stream.data = c.portData[headers.Get(api.PortHeader)]
+	}
 
 	return stream, nil
 }

